src/protocol/context/semi_honest.rs

```diff
@@ -194,6 +214,31 @@ impl<'a, F: ExtendableField> UpgradedContext<F> for Upgraded<'a, F> {
         Ok(x)
     }
 
+    // Moved from "context/mod.rs" because we want to use the local `UpgradeContext`.
+    // `upgrade` for semi-honest used to be narrowed but never executed. That causes
+    // `Compact` gate to panic because it doesn't know anything about state transitions
+    // that are not executed. To workaround this, we use a dummy step to narrow the
+    // context to let `Compact` gate know that it could be ignored.
```

Comment:
do we still need to add `upgrade` to the list of "static" mappings after thins change?

Reply:
That is correct.

I haven't made the change we tried when we were in London. That turned out to not work because there are more narrows after `upgrade` (in `UpgradeToMalicious` implementations before the no-op functions `UpgradedContext::upgrade_[one|sparse]` are called).